include/RandLAPACK/comps/orth.hh

```diff
@@ -21,99 +19,114 @@ class Stabilization {
+        }
+};
+
+
+template <typename T>
+class HQRQ : public Stabilization<T> {
+    public:
+        std::vector<T> tau;
+        bool cond_check;
+        bool verbosity;
+
+        // Constructor
+        HQRQ(bool c_check, bool verb) {
+            cond_check = c_check;
+            verbosity = verb;
+        };
+
+        int hqrq(
             int64_t m,
             int64_t n,
             std::vector<T>& A,
             std::vector<T>& tau
         );
 
-        int GEQR(
+        int call(
+            int64_t m,
+            int64_t k,
+            std::vector<T>& Q
+        ){
+            return hqrq(m, k, Q, this->tau);;
+        }
+};
+
+#if !defined(__APPLE__)
+template <typename T>
+class GEQR : public Stabilization<T> {
+    public:
+        std::vector<T> tvec;
+        bool cond_check;
+        bool verbosity;
+
+        // Constructor
+        GEQR(bool c_check, bool verb) {
+            cond_check = c_check;
+            verbosity = verb;
+        };
+
+        int geqrq(
             int64_t m,
             int64_t n,
             std::vector<T>& A,
             std::vector<T>& tvec
         );
 
-        // Control of Orth types calls.
         int call(
             int64_t m,
             int64_t k,
             std::vector<T>& Q
         ){
-            // Default
-            int termination = 0;
-            switch(this->decision_orth) 
-            {
-                case use_CholQRQ:
-                    if(this->chol_fail) {
-                        termination = HQRQ(m, k, Q, this->tau);
-                    } else {
-                        //Conventionally, we may call it twice for better orthogonality. Practically, the 2nd call makes no difference.
-                        if(CholQRQ(m, k, Q)) {
-                                termination = HQRQ(m, k, Q, this->tau);
-                        }
-                        //termination = CholQRQ(m, k, Q);
-                    }
-                    break;
-                case use_HQRQ:
-                    termination = HQRQ(m, k, Q, this->tau);
-                    break;
-                case use_GEQR: 
-#if !defined(__APPLE__)
-                    termination = GEQR(m, k, Q, this->tvec);
-#else
-                    throw(1);  // GEQR not available on macOS; must use HQR.
-#endif
-            }
-            return termination;
+            return geqrq(m, k, Q, this->tvec);
         }
 };
+#endif
 
 template <typename T>
-class Stab : public Orth<T>
+class PLUL : public CholQRQ<T>
 {
     public:
         std::vector<int64_t> ipiv;
-        decision_orth_stab decision_stab;
         
         // Constructor
-        Stab(decision_orth_stab dec, bool c_check, bool verb) : Orth<T>::Orth(dec, c_check, verb) {
+        PLUL(bool c_check, bool verb) : CholQRQ<T>::CholQRQ(c_check, verb) {
             this->cond_check = c_check;
             this->verbosity = verb;
-            decision_stab = dec;
             this->chol_fail = false; 
         };
```

Comment:
This is a leftover issue from when the class "Stab" was subclass "Orth" so that CholQRQ could be used for both stabilization and orthogonalization. This is now fixed.